qutebrowser/misc/sql.py

```diff
@@ -0,0 +1,178 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides access to an in-memory sqlite database."""
+
+from PyQt5.QtCore import QObject, pyqtSignal
+from PyQt5.QtSql import QSqlDatabase, QSqlQuery
+
+from qutebrowser.utils import log
+
+import collections
+
+
+def init():
+    """Initialize the SQL database connection."""
+    database = QSqlDatabase.addDatabase('QSQLITE')
+    # In-memory database, see https://sqlite.org/inmemorydb.html
+    database.setDatabaseName(':memory:')
+    if not database.open():
+        raise SqlException("Failed to open in-memory sqlite database")
```

Comment:
Maybe add `database.lastError().text()` here? See [QSqlDatabase::lastError](http://doc.qt.io/qt-5/qsqldatabase.html#lastError) and [QSqlError](http://doc.qt.io/qt-5/qsqlerror.html).